ocaml/xcp-rrdd/bin/rrdp-iostat/rrdp_iostat.ml

```diff
@@ -445,46 +452,36 @@ let exec_tap_ctl () =
        update_vdi_to_vm_map ()
   ) ;
   previous_map := pid_and_minor_to_sr_and_vdi ;
-  minor_to_sr_and_vdi
+  sr_and_vdi_to_minor
 
-let minor_of_tdX_unsafe tdX =
+(* Get the minor number of the kernel tapdev device *)
+let minor_of_tapdev_unsafe tapdev =
   int_of_string
     (Unixext.file_lines_fold
        (fun acc l -> acc ^ List.nth (Xstringext.String.split ':' l) 1)
```

Comment:
String concatenations with `^` are expensive. `List.nth` is could be solved with pattern matching; its' ok here because `1` is a small number.